vpr/src/place/analytical_placement/analytical_placement_flow.cpp

```diff
@@ -51,26 +53,19 @@ void run_analytical_placement_flow() {
     // Using the placement from the SA to get the fixed block locations
     std::set<AtomBlockId> fixed_blocks;
     std::map<AtomBlockId, double> fixed_blocks_x, fixed_blocks_y;
+    const VprConstraints constraints = g_vpr_ctx.mutable_floorplanning().constraints;
```

Comment:
I wouldn't use the mutable version of the floorplanning since you are not modifying it anyways. Can you use the non-mutable version?

Also this should get the value of the constraints: `const VprConstraints& constraints`. I am not sure if this actually improves performance, but it makes it more clear that we do not want to create a copy of constraints.

Reply:
done